app/group/tests/test_membership_model.py

```diff
@@ -82,3 +87,53 @@ def test_swap_leader_does_not_create_two_leaders(group):
 def test_on_delete_membership_history_is_created(membership):
     membership.delete()
     assert MembershipHistory.objects.get(start_date=membership.created_at)
+
+
+@pytest.mark.django_db
+def test_on_swap_creates_hs_membership_with_no_leader(hs):
+    """"
+        Tests that if you promote as user to leader of a subgroup with no leader,
+        the user is automaticly added to the HS group
+    """
+    group = GroupFactory(type=GroupType.SUBGROUP)
+    membership = MembershipFactory(membership_type=MembershipType.MEMBER, group=group)
+    membership.membership_type = MembershipType.LEADER
+    membership.save()
+    membership.refresh_from_db()
+    assert membership.membership_type == MembershipType.LEADER
+    assert Membership.objects.get(user=membership.user, group=hs)
+
+
+@pytest.mark.django_db
+def test_on_swap_creates_hs_membership_with_leader(hs):
+    """
+    Tests that if you promote as user to leader of a subgroup,
+    the user is automaticly added to the HS group
+    """
+    group = GroupFactory(type=GroupType.SUBGROUP)
+    membership = MembershipFactory(membership_type=MembershipType.MEMBER, group=group)
+    membership_leader = MembershipFactory(
+        membership_type=MembershipType.LEADER, group=group
+    )
+    membership.membership_type = MembershipType.LEADER
+    membership.save()
+    membership.refresh_from_db()
+    membership_leader.refresh_from_db()
+    assert membership.membership_type == MembershipType.LEADER
+    assert membership_leader.membership_type == MembershipType.MEMBER
+    assert Membership.objects.get(user=membership.user, group=hs)
+
+
+@pytest.mark.django_db
+def test_on_remove_leader_without_swap_removes_hs(hs):
+    """
+    Tests that if you demote as user from leader to member of a subgroup,
+    the user is automaticly removed from the HS group
+    """
+    group = GroupFactory(type=GroupType.SUBGROUP)
+    membership = MembershipFactory(membership_type=MembershipType.LEADER, group=group)
+    membership.membership_type = MembershipType.MEMBER
+    membership.save()
+    membership.refresh_from_db()
+    assert membership.membership_type == MembershipType.MEMBER
+    assert not len(Membership.objects.filter(user=membership.user, group=hs))
```

Comment:
Her kan vi bruke litt Django magi: 
```suggestion
    assert not membership.user.membership_set.filtergroup=hs).exists()
```

Hadde vi brukt `related_name=memberships` som argument [i User](https://github.com/TIHLDE/Lepton/blob/c4844b8fcd3fd329ca0051e5db3bfb965ffec537/app/group/models/membership.py#L17) kunne den blitt enda mer lesbar: 
`assert not membership.user.memberships.filter(group=hs).exists()`

Dette kan brukes i alle øvrige tester også hvis du ønsker det :)